Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,140 @@
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets)">
+    /// [BigQuery Datasets]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcbqDataset")]
+    public class NewGcbqDataset : GcbqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "premade")]
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The dataset object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = "premade")]
+        public Dataset Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatasetId for the datset reference object.  Must be unique within the project, and must be 1024 
+        /// characters or less.  It must only contain letters (a-z, A-Z), numbers (0-9), or underscores (_).
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = "builder")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "friendly name" field for the dataset.  Used for a descriptive name for the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "description" field for the dataset.  Used for a user-friendly description of the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The default lifetime for tables in the dataset to exist in ms.  Minimum is 3600000 (1hr).  
+        /// When the expiration time is reached, the table will be automatically deleted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "builder")]
+        public int Timeout { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Dataset newData;
+            // Check whether they are passing in a completed Dataset object or are building one with params.
+            if (Dataset != null)
```

Comment:
You should check the name of the parameterset instead. See [this](https://github.com/GoogleCloudPlatform/google-cloud-powershell/blob/c8295d5634c02caf26f3c6db58dd8f87a5dd4be2/Google.PowerShell/Compute/GceAddressCmdlets.cs#L97) for example.